cli/smartcontract/generate.go

```diff
@@ -0,0 +1,350 @@
+			Usage: "Output of the compiled contract",
+		},
+		cli.StringFlag{
+			Name:  "hash",
+			Usage: "Smart-contract hash",
+		},
+		cli.StringSliceFlag{
+			Name: "override",
+			Usage: `Override type for methods. Should have form <method>.<name>=<type>.
+Type should be fully-qualified name of the type. For return use <method>=<type>'`,
+		},
+		cli.StringSliceFlag{
+			Name: "callflag",
+			Usage: `Override call flags. Should have form <method>=<callflag>.
+Call flag can be one of ` + strings.Join(callFlags, ","),
+		},
+	},
+}
+
+func scTypeToGo(typ smartcontract.ParamType) string {
+	switch typ {
+	case smartcontract.AnyType:
+		return "interface{}"
+	case smartcontract.BoolType:
+		return "bool"
+	case smartcontract.IntegerType:
+		return "int"
+	case smartcontract.ByteArrayType:
+		return "[]byte"
+	case smartcontract.StringType:
+		return "string"
+	case smartcontract.Hash160Type:
+		return "interop.Hash160"
+	case smartcontract.Hash256Type:
+		return "interop.Hash256"
+	case smartcontract.PublicKeyType:
+		return "interop.PublicKey"
+	case smartcontract.SignatureType:
+		return "interop.Signature"
+	case smartcontract.ArrayType:
+		return "[]interface{}"
+	case smartcontract.MapType:
+		return "map[string]interface{}"
+	case smartcontract.InteropInterfaceType:
+		return "interface{}"
+	case smartcontract.VoidType:
+		return ""
+	default:
+		panic("unreachable")
+	}
+}
+
+// contractGenerateWrapper deploys contract.
+func contractGenerateWrapper(ctx *cli.Context) error {
+	manifestFile := ctx.String("manifest")
+	if len(manifestFile) == 0 {
+		return cli.NewExitError(errNoManifestFile, 1)
+	}
+
+	manifestBytes, err := ioutil.ReadFile(manifestFile)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to read manifest file: %w", err), 1)
+	}
+
+	m := &manifest.Manifest{}
+	err = json.Unmarshal(manifestBytes, m)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to restore manifest file: %w", err), 1)
+	}
+
+	s, err := Generate(ctx, m)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during generation: %w", err), 1)
+	}
+
+	err = ioutil.WriteFile(ctx.String("out"), []byte(s), os.ModePerm)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during write: %w", err), 1)
+	}
+	return nil
+}
+
+func parseCallFlags(ctx *cli.Context) (map[string]string, error) {
+	flags := ctx.StringSlice("callflag")
+	result := make(map[string]string, len(flags))
+loop:
+	for i := range flags {
+		ss := strings.SplitN(flags[i], "=", 2)
+		if len(ss) == 1 {
+			return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+		}
+		for i := range callFlags {
+			if ss[1] == callFlags[i] {
+				result[ss[0]] = ss[1]
+				continue loop
+			}
+		}
+		return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+	}
+	return result, nil
+}
+
+type override struct {
+	Package  string
+	TypeName string
+}
+
+func parseOverrides(ctx *cli.Context) (map[string]override, error) {
+	overrides := ctx.StringSlice("override")
```

Comment:
I am not sure it is worth to do, it is a oneshot command and all such arguments refer to a single place in code. Maybe we could extend our config file to support custom annotations, though? This way is could be kept in sync with the contract development and binging generation will be just `-m manifest -c config.yml`.

cc @roman-khimov

Reply:
Not sure that the same config we use for compilation suits here because e.g. we already have contract name inside the manifest, but we definitely can make use of config fields like overloads. I'd say it should be a separate manually constructed configuration file (not tied to the config that's used for contract compilation) because we may want to generate binding for a contract we don't have standard `config.yml` (e.g. compiled by the C# compiler).

Reply:
Ok, so we can have a separate config which can also be optionally emitted by compiler.

Reply:
Created a task for compiler emit #2358 .